pkg/apis/cloudcredential/v1/credentialsrequest_types.go

```diff
@@ -131,6 +131,10 @@ const (
 	// CredentialsDeprovisionFailure is true whenever there is an error when trying
 	// to clean up any previously-created cloud resources
 	CredentialsDeprovisionFailure CredentialsRequestConditionType = "CredentialsDeprovisionFailure"
+	// OtherInfrastrucutre is true when the CredentialsRequest's ProviderSpec is for
+	// a different infrastructure platform than what the cluster has been
+	// deployed to.
+	OtherInfrastructure CredentialsRequestConditionType = "OtherInfrastructure"
```

Comment:
WDYT about renaming the condition to Ignored, reason "ProviderMismatch" or "InfrastructureMismatch" or "PlatformMismatch"

Reply:
We should also expand on the doc comment just to note that this is normal in that we deploy creds for all clouds in the release image and only act on those if the cloud infra matches.

Reply:
yes, that makes way more sense. done.